apps/web/src/app/(dashboard)/settings/profile/delete-account-dialog.tsx

```diff
@@ -103,17 +109,44 @@ export const DeleteAccountDialog = ({ className, user }: DeleteAccountDialogProp
                   , along with all of your completed documents, signatures, and all other resources
                   belonging to your Account.
                 </DialogDescription>
+                {!hasTwoFactorAuthentication && (
+                  <DialogDescription>
+                    Please type <span className="font-semibold">{username}</span> to confirm.
+                  </DialogDescription>
+                )}
               </DialogHeader>
 
+              {!hasTwoFactorAuthentication && (
+                <div className="mt-4">
+                  <Input
+                    type="text"
+                    value={enteredUsername}
+                    onChange={(e) => setEnteredUsername(e.target.value)}
+                    onPaste={(e) => e.preventDefault()}
```

Comment:
To make sure user cannot paste directly onto the input box.

Reply:
We never really want to disable paste, I always copy paste these kinds of confirmations. The extra step of having to enter it either manually or via copy + paste makes it intentional enough to avoid doing it accidentally 😄